### PR TITLE
[CWS] upgrade opa rego to v0.37.2

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR upgrades the rego evaluator dependency.

This bump was tested on the `security-monitoring` rego testsuite

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by paulcacheux** [comment]
@mx-psi I asked a review from you because of the changes to other dependencies that this bump brings. One dependency in particular seems to be the prometheus client. Looking at the changelog it seems that v1.12.0 actually adds support for Go 1.17 so maybe we actually need it ?

**Comment by mx-psi** [comment]
I don't know which Prometheus features we use in the Agent, that is a question for Agent Core. If we use the feature update in prometheus/client_golang#955 I think we should update indeed. In any case, I think this PR leaves us better (or at least not worse) than we currently are, so I will approve it!

**Review by mx-psi — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -149,13 +149,13 @@ require (
...
 	github.com/olekukonko/tablewriter v0.0.5
-	github.com/open-policy-agent/opa v0.35.0
+	github.com/open-policy-agent/opa v0.37.2
 	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/resourcetotelemetry v0.43.0
 	github.com/opencontainers/runtime-spec v1.0.3-0.20210326190908-1c3f411f0417
 	github.com/openshift/api v0.0.0-20190924102528-32369d4db2ad
 	github.com/patrickmn/go-cache v2.1.0+incompatible
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.11.0
+	github.com/prometheus/client_golang v1.12.0
```
Comment (hush-hush) [inline go.mod:158]: should we go for 1.12.1 ?

Comment (paulcacheux) [reply go.mod:158]: This is purely the result of running `inv -e tidy-all`. But yes I think we can, let me push a commit

**Review by hush-hush — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

**Pull request merged**